lib/events.js

```diff
@@ -240,23 +243,25 @@ EventEmitter.prototype.removeListener =
 
 EventEmitter.prototype.removeAllListeners =
     function removeAllListeners(type) {
-      var key, listeners;
+      var listeners, events;
 
-      if (!this._events)
+      if (!(events = this._events))
         return this;
 
       // not listening for removeListener, no need to emit
-      if (!this._events.removeListener) {
+      if (!events.removeListener) {
         if (arguments.length === 0)
           this._events = {};
-        else if (this._events[type])
-          delete this._events[type];
+        else if (events[type])
+          delete events[type];
         return this;
       }
 
       // emit removeListener for all listeners on all events
       if (arguments.length === 0) {
-        for (key in this._events) {
+        var keys = Object.keys(events);
```

Comment:
I suppose Object.keys() could have a negative impact on EventEmitter objects with many events because of the array it creates (instead of iterating over an object like for..in does.)  Probably an uncommon case but it might be good to capture it in a benchmark.